ludwig/automl/utils.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+
+import GPUtil
+import psutil
+from dataclasses_json import LetterCase, config, dataclass_json
+from pandas import Series
+
+
+@dataclass_json(letter_case=LetterCase.CAMEL)
+@dataclass
+class FieldInfo:
+    name: str
+    dtype: str
+    key: str = None
+    distinct_values: int = 0
+    nonnull_values: int = 0
+    avg_words: int = None
+
+
+def get_avg_words(field: Series) -> int:
+    # sample a subset if dataframe is large
+    if len(field) > 5000:
+        field = field.sample(n=5000, random_state=40)
+    unique_entries = field.unique()
+    avg_words = Series(unique_entries).str.split().str.len().mean()
+    return avg_words
+
+
+def get_available_resources():
```

Comment:
For Ray, we can use the resource info provided by Ray to obtain this:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/backend/ray.py#L49